src/net_processing.cpp

```diff
@@ -2566,6 +2572,18 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         return;
     }
 
+    if (msg_type == NetMsgType::BLOCKRELAY) {
+        if (pfrom.fSuccessfullyConnected) {
```

Comment:
Out of curiosity, why must `BLOCKRELAY` be negotiated between `VERSION` and `VERACK`? The only reason I can think of is to avoid allocating (`m_tx_relay`, `m_addr_known`) altogether instead of allocating, then de-allocating. Is that correct?